main.py

```
@@ -604,0 +1,1133 @@
+        logging.StreamHandler(),
+        logging.FileHandler("markdown_scraper.log", mode="a"),
+    ],
+)
+
+logger = logging.getLogger("markdown_scraper")
+
+
+class RequestCache:
+    """Simple cache for HTTP requests to avoid repeated network calls."""
+
+    def __init__(self, cache_dir: str = ".request_cache", max_age: int = 3600):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)